sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/QueueServiceAsyncAPITests.groovy

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue.spock
+
+import com.azure.storage.queue.models.*
+import reactor.test.StepVerifier
+import spock.lang.Unroll
+
+import static org.junit.Assert.assertFalse
+import static org.junit.Assert.assertNotNull
+
+class QueueServiceAsyncAPITests extends APISpec {
+
+    def "Get queue client from queue service async client"() {
+        given:
+        def queueName = testResourceName.randomName("queue", 16)
+        when:
+        def queueAsyncClient = primaryQueueServiceAsyncClient.getQueueAsyncClient(queueName)
+        def enqueueMsgVerifier = StepVerifier.create(queueAsyncClient.enqueueMessage("Expecting an exception"))
+        then:
+        assertNotNull(queueAsyncClient)
+        enqueueMsgVerifier.verifyErrorSatisfies {
+            QueueTestHelper.assertExceptionStatusCodeAndMessage(it, 404, "QueueNotFound")
+        }
+    }
+
+    def "Create queue from queue service async client"() {
+        given:
+        def queueName = testResourceName.randomName("queue", 16)
+        when:
+        def createQueueVerifier = StepVerifier.create(primaryQueueServiceAsyncClient.createQueue(queueName))
+        def enqueueMessageVerifier = StepVerifier.create(primaryQueueServiceAsyncClient.getQueueAsyncClient(queueName).enqueueMessage("Testing service client creating a queue"))
+        then:
+        createQueueVerifier.assertNext {
+            QueueTestHelper.assertResponseStatusCode(it, 201)
+        }.verifyComplete()
+        enqueueMessageVerifier.assertNext {
+            QueueTestHelper.assertResponseStatusCode(it, 201)
+        }.verifyComplete()
+    }
+
+    def "Create queues with duplicate name from queue service async client"() {
+        given:
+        def queueName = testResourceName.randomName("queue", 16)
+        primaryQueueServiceAsyncClient.createQueue(queueName).block()
+        when:
+        def createQueueVerifier = StepVerifier.create(primaryQueueServiceAsyncClient.createQueue(queueName))
+        then:
+        createQueueVerifier.assertNext {
+            QueueTestHelper.assertResponseStatusCode(it, 204)
+        }.verifyComplete()
+    }
+
+    @Unroll
+    def "Create queue with invalid name from queue service async client"() {
+        when:
+        def createQueueVerifier = StepVerifier.create(primaryQueueServiceAsyncClient.createQueue(queueName))
+        then:
+        createQueueVerifier.verifyErrorSatisfies {
+            QueueTestHelper.assertExceptionStatusCodeAndMessage(it, statusCode, errMesage)
+        }
+        where:
+        queueName      | statusCode | errMesage
+        "a_b"          | 400        | "InvalidResourceName"
```

Comment:
Instead of using a String for the errMessage (type there too), could we use the StorageErrorCode enum.

Reply:
Yes, I am going to have them move over to common module.

Reply:
Don't move the StorageErrorCode class to common, they are different for each SDK (Blobs, Files, and Queues) and they are autogen code.

Reply:
I did not make myself clear. I mean it is there after your PR check in. 
I am going to use the StorageErrorCode